core/src/main/scala/org/bitcoins/core/psbt/PSBT.scala

```diff
@@ -413,6 +476,32 @@ object PSBT extends Factory[PSBT] {
       case Some(bytes) => fromBytes(bytes)
     }
   }
+
+  def fromUnsignedTx(unsignedTx: Transaction): PSBT = {
+    val globalMap = GlobalPSBTMap(
+      Vector(GlobalPSBTRecord.UnsignedTransaction(unsignedTx)))
+    val inputMaps = unsignedTx.inputs.map(_ => InputPSBTMap.empty).toVector
+    val outputMaps = unsignedTx.outputs.map(_ => OutputPSBTMap.empty).toVector
+
+    PSBT(globalMap, inputMaps, outputMaps)
+  }
+
+  def fromUnsignedTxAndInputs(
+      unsignedTx: Transaction,
+      spendingInfos: Vector[UTXOSpendingInfo])(
+      implicit ec: ExecutionContext): Future[PSBT] = {
+    require(spendingInfos.length == unsignedTx.inputs.length,
+            "Must have a UTXOSpendingInfo for every input")
```

Comment:
Could validate that the `spendingInfos` correspond to inputs in the `unsignedTx` as well